credhub-uaa/credhub-uaa.yml

```
@@ -0,0 +1,178 @@
+---
+name: ((deployment_name))
+instance_groups:
+- name: credhub-uaa
+  azs:
+  - z1
+  instances: 1
+  vm_type: uaa
+  persistent_disk_type: large
+  stemcell: xenial
+  networks:
+  - name: private
+  vm_extensions:
+     release: uaa
+    properties:
+      encryption:
+        active_key_label: 'key-1'
+        encryption_keys:
+          - label: 'key-1'
+            passphrase: "((uaa_encryption_key))"
+      login:
+        saml:
+          serviceProviderCertificate: "((uaa_login_saml.certificate))"
+          serviceProviderKey: "((uaa_login_saml.private_key))"
+        branding:
+          company_name: "Concourse"
+      uaa:
+        clients:
+          admin:
+            authorized-grant-types: client_credentials
+            scope: uaa.none
+            authorities: uaa.admin,clients.read,clients.write,clients.secret,scim.read,scim.write,clients.admin
+            secret: "((uaa_admin_client_secret))"
+          credhub_cli:
+            authorized-grant-types: password,refresh_token
+            authorities: uaa.none
+            scope: credhub.read,credhub.write
+            secret: "" # credhub expects this to be empty
+            access-token-validity: 120
+            refresh-token-validity: 1800
+            override: true
+          concourse_client:
+            authorized-grant-types: client_credentials
+            authorities: credhub.read,credhub.write
+            scope: credhub.read,credhub.write
+            secret: "((concourse_credhub_client_secret))"
+            access-token-validity: 120
+            refresh-token-validity: 1800
+            override: true
+        jwt:
+          policy:
+            active_key_id: key-1
+            keys:
+              key-1:
+                signingKey: "((uaa_jwt_signing_key.private_key))"
+        scim:
+          users:
+            - name: admin
+              password: "((cf_admin_password))"
+              groups:
+                - uaa.admin
+            - name: credhub
+              password: "((credhub_user_password))"
+              groups:
+                - credhub.read
+                - credhub.write
```

Needs the acl permissions. Not 100% sure these exist, should ask the credhub team on slack. I can't find any docs about these groups being in uaa, only found out about them in the credhub API.
```suggestion
                - credhub.write
                - credhub.read_acl
                - credhub.write_acl
```